splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/sequence/AbstractSequence.java

```diff
@@ -52,6 +52,10 @@ public long getNext() throws StandardException{
         return currPosition.getAndAdd(incrementSteps);
     }
 
+    public long peekAtCurrentValue() throws StandardException {
+        return currPosition.get();
```

Comment:
If the current block is consumed this will return a wrong result. We should perform the same check as in getNext() and allocate a new block if necessary.